### DISCUSSION TITLE
Wrong colors second led strip in multi-segment setup

### DISCUSSION THREAD

**Discussion by myswa87** [awawa-dev/HyperSerialESP32#32] [Q&A]
Hello Awawa,

I find it very challeging to get two correctly working led strips with the multi-segment option.

I've got a test line-up with a ESP32 MH-ET Live (CH9102) and two ws2812b led strips.
Now the problem is that the second led strip always shows different colors, always three colors and this repeats over the whole second led strip (see attached image)

I used the following settings in the platformio.ini:

build_flags = -DSERIALCOM_SPEED=4000000

build_flags = -DNEOPIXEL_RGB -DDATA_PIN=2 ${env.build_flags} -DSECOND_SEGMENT_START_INDEX=27 -DSECOND_SEGMENT_DATA_PIN=4 -DSECOND_SEGMENT_REVERSED

In the HyperHDR logging I can see that the adalight is recognized:

[LEDDEVICE_ADALIGHT] DETECTED DEVICE USING HYPERSERIALESP8266/HYPERSERIALESP32 FIRMWARE (Welcome! Awa driver 9. SECONDSEGMENTSTARTINDEX = 27) at 85 msec

I've added my HyperHDR configuration backup in the attached .zip file.
The link to my own fork is: https://github.com/myswa87/HyperSerialESP32

Do you have any ideas where it goes wrong? I already tried:
- different ESP boards
- choose different GPIO's
- different cables.
- clean HyperHDR installation on both Windows and Linux.
- tried only with a background effect color

I hope to hear from you, thanks in advance.

[HyperHDR-19.0.0.0-Backup-2023-03-30.json.zip](https://github.com/awawa-dev/HyperSerialESP32/files/11114840/HyperHDR-19.0.0.0-Backup-2023-03-30.json.zip)
[led_strips](https://user-images.githubusercontent.com/129425838/228931822-242d8547-40e1-4390-8023-d56ee4f034f9.jpg)

**Comment by awawa-dev** [comment]
Hi
If for any configuration witch such symptoms (first segment is working for you and the second segment isn't), did you try to switch these segment data cables on esp32 pinout without any software changes? In such situation did the second segment start to work and the first stop (repeating colors)?

**Comment by awawa-dev** [comment]
I can confirm there was a problem:
https://github.com/myswa87/HyperSerialESP32/actions/runs/4567413788/jobs/8061169966#step:7:378
![obraz](https://user-images.githubusercontent.com/69086569/228950575-a13f0e5b-f01e-4e8d-8f4a-84b3a716fbb0.png)

as you can see the second segment due to my typo was set to RGBW instead of RGB for ws2812b.
I fixed it now. Please update your fork to latest upstream and recompile the firmware.

**Reply by phil535** [reply]
Thanks for the fix, it works now on master 5687e8e.

I also had this issue in release v9.0.0.0.
